pallets/oracle-data-collection/src/lib.rs

```diff
@@ -0,0 +1,496 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Oracle pallet to collect and aggregate oracle values.
+//!
+//! The collection admin configures the collection allowing a list of feeders.
+//!
+//! Later, the updating a collection will collect all values based on the admin
+//! configuration of the collection. The resulting collection is optimized to
+//! iterate though all values in jus one read.
+//!
+//! This pallet is not fed with external values, you need to configure a
+//! `ValueProvider` to provide it with oracle values.
```

Comment:
Nit: Let's make the assumption more explicit:

```suggestion
//! ### Assumptions
//!
//! This pallet is not fed with external values. You need to configure a
//! `ValueProvider` to provide it with oracle values.
```

Reply:
Agree!